lib/ash_graphql.ex

```diff
@@ -283,6 +291,14 @@ defmodule AshGraphql do
     |> Enum.uniq_by(& &1.identifier)
   end
 
+  defp enum_type_description(type, value) do
+    if Spark.implements_behaviour?(type, Ash.Type.Enum) do
+      type.description(value)
+    else
```

Comment:
Ah, damn. To handle types that don't `use AshGraphql.Type` we should add `if function_exported?` around this as well.

Reply:
@smt116 are you on 3.0 packages yet?

Reply:
Actually... 🤔 this shouldn't be an issue if Ash is also upgraded. But I'll add the check anyway. Since `use Ash.Type.Enum` would define the `description` function. Are you overriding dependency versions or anything like that?

Reply:
No. Working on upgrading everything to latest 2.x and then to 3.x. I'm already adding `description/1` to all our enums.

Reply:
Currently we are in the following state

```
  "ash": {:hex, :ash, "2.19.4", "18ce485f4d1aaedcc7b09c3b74031fc734732edbef311ccbc6263305e78bd974", [:mix], [{:comparable, "~> 1.0", [hex: :comparable, repo: "hexpm", optional: false]}, {:decimal, "~> 2.0", [hex: :decimal, repo: "hexpm", optional: false]}, {:earmark, "~> 1.4", [hex: :earmark, repo: "hexpm", optional: false]}, {:ecto, "~> 3.7", [hex: :ecto, repo: "hexpm", optional: false]}, {:ets, "~> 0.8", [hex: :ets, repo: "hexpm", optional: false]}, {:jason, ">= 1.0.0", [hex: :jason, repo: "hexpm", optional: false]}, {:picosat_elixir, "~> 0.2", [hex: :picosat_elixir, repo: "hexpm", optional: false]}, {:plug, ">= 0.0.0", [hex: :plug, repo: "hexpm", optional: true]}, {:spark, ">= 1.1.50 and < 2.0.0-0", [hex: :spark, repo: "hexpm", optional: false]}, {:stream_data, "~> 0.6", [hex: :stream_data, repo: "hexpm", optional: false]}, {:telemetry, "~> 1.1", [hex: :telemetry, repo: "hexpm", optional: false]}], "hexpm", "59de46ebdd18fcf7c5caa38e1b2e587e501ccc2e5187d6c5c4d8e3e50dbc1b70"},
  "ash_appsignal": {:hex, :ash_appsignal, "0.1.3", "90c0ed1b3aedd18b892352d2a7632641f9ef8f2d3b04caec2bdc810cb8900524", [:mix], [{:appsignal, "~> 2.0", [hex: :appsignal, repo: "hexpm", optional: false]}, {:ash, "~> 3.0", [hex: :ash, repo: "hexpm", optional: false]}], "hexpm", "43b1691c9e83a27a355ee1d2c047a79e18adac99eb7e8f1a532feebfe2a07a14"},
  "ash_authentication": {:hex, :ash_authentication, "3.12.4", "7f1535c55d8f5896b33960daaf1f046d365b830be5292d94a72eef1a8be918bf", [:mix], [{:ash, ">= 2.5.11 and < 3.0.0-0", [hex: :ash, repo: "hexpm", optional: false]}, {:ash_postgres, "~> 1.5.1", [hex: :ash_postgres, repo: "hexpm", optional: true]}, {:assent, ">= 0.2.8 and < 1.0.0-0", [hex: :assent, repo: "hexpm", optional: false]}, {:bcrypt_elixir, "~> 3.0", [hex: :bcrypt_elixir, repo: "hexpm", optional: false]}, {:castore, "~> 1.0", [hex: :castore, repo: "hexpm", optional: false]}, {:finch, "~> 0.18.0", [hex: :finch, repo: "hexpm", optional: false]}, {:jason, "~> 1.4", [hex: :jason, repo: "hexpm", optional: false]}, {:joken, "~> 2.5", [hex: :joken, repo: "hexpm", optional: false]}, {:plug, "~> 1.13", [hex: :plug, repo: "hexpm", optional: false]}, {:spark, ">= 1.1.39 and < 2.0.0-0", [hex: :spark, repo: "hexpm", optional: false]}], "hexpm", "771827fabcba017a0c36858a50533263fb30a77ffbecbbec92a2733412e6a1c1"},
  "ash_authentication_phoenix": {:hex, :ash_authentication_phoenix, "1.9.4", "bd7b4c8597e1479f385d93f83bb5b9a486e1ad3e77b0ed3f682c482e549bfa77", [:mix], [{:ash, "~> 2.2", [hex: :ash, repo: "hexpm", optional: false]}, {:ash_authentication, ">= 3.11.9 and < 4.0.0-0", [hex: :ash_authentication, repo: "hexpm", optional: false]}, {:ash_phoenix, "~> 1.1", [hex: :ash_phoenix, repo: "hexpm", optional: false]}, {:bcrypt_elixir, "~> 3.0", [hex: :bcrypt_elixir, repo: "hexpm", optional: false]}, {:jason, "~> 1.0", [hex: :jason, repo: "hexpm", optional: false]}, {:phoenix, "~> 1.6", [hex: :phoenix, repo: "hexpm", optional: false]}, {:phoenix_html, "~> 4.0", [hex: :phoenix_html, repo: "hexpm", optional: false]}, {:phoenix_html_helpers, "~> 1.0", [hex: :phoenix_html_helpers, repo: "hexpm", optional: false]}, {:phoenix_live_view, "~> 0.18", [hex: :phoenix_live_view, repo: "hexpm", optional: false]}, {:phoenix_view, "~> 2.0", [hex: :phoenix_view, repo: "hexpm", optional: false]}, {:slugify, "~> 1.3", [hex: :slugify, repo: "hexpm", optional: false]}], "hexpm", "a37966ceb75423cbebbcc01224b4bc4e4904bcf21d7468498994b40d43b4b254"},
  "ash_graphql": {:hex, :ash_graphql, "0.28.1", "a75518317ed2590891b144dbef450b0aa8f5cc714491d13016a994519d9b2ec4", [:mix], [{:absinthe, "~> 1.7", [hex: :absinthe, repo: "hexpm", optional: false]}, {:absinthe_plug, "~> 1.4", [hex: :absinthe_plug, repo: "hexpm", optional: false]}, {:ash, "~> 2.17", [hex: :ash, repo: "hexpm", optional: false]}, {:jason, "~> 1.2", [hex: :jason, repo: "hexpm", optional: false]}], "hexpm", "ec6a06819f02ff34e1527a29536998d5c0f8b9317f2b27ffec80d5638d58d940"},
  "ash_json_api": {:hex, :ash_json_api, "0.34.2", "21a1f935d1208d7f419f08cb44ae379ffa9919dc4860e6bbc6e7499762986e7e", [:mix], [{:ash, ">= 2.9.24 and < 3.0.0-0", [hex: :ash, repo: "hexpm", optional: false]}, {:jason, "~> 1.1", [hex: :jason, repo: "hexpm", optional: false]}, {:json_xema, "~> 0.4.0", [hex: :json_xema, repo: "hexpm", optional: false]}, {:open_api_spex, "~> 3.16", [hex: :open_api_spex, repo: "hexpm", optional: true]}, {:plug, "~> 1.11", [hex: :plug, repo: "hexpm", optional: false]}], "hexpm", "620658e495ac745807d8eab0e752836f44e1368c98c7beaad5d4c2bd8c286cf4"},
  "ash_oban": {:hex, :ash_oban, "0.2.2", "85a1a6876c68e331762752c1ca37420fcbb7a3f9c8daa2136675c303463347ed", [:mix], [{:ash, ">= 2.9.27 and < 3.0.0-0", [hex: :ash, repo: "hexpm", optional: false]}, {:oban, "~> 2.15", [hex: :oban, repo: "hexpm", optional: false]}, {:postgrex, "~> 0.17.4", [hex: :postgrex, repo: "hexpm", optional: false]}, {:spark, ">= 1.1.3", [hex: :spark, repo: "hexpm", optional: false]}], "hexpm", "6f8af11a4290a4400fa4f441d5b2b20714f598feff2811203a75f39bd35eeb59"},
  "ash_phoenix": {:hex, :ash_phoenix, "1.3.4", "8b38b5822cb68e1b8a1e9ccf947331318fcb7e8df53da841b229b3305aa6c0dd", [:mix], [{:ash, "~> 2.16", [hex: :ash, repo: "hexpm", optional: false]}, {:phoenix, "~> 1.5.6 or ~> 1.6", [hex: :phoenix, repo: "hexpm", optional: false]}, {:phoenix_html, "~> 4.0", [hex: :phoenix_html, repo: "hexpm", optional: false]}, {:phoenix_live_view, "~> 0.20.3", [hex: :phoenix_live_view, repo: "hexpm", optional: false]}], "hexpm", "3d13fe54887c84cd89c7106aa05b7bbc87498b2da53e055b21ae21173fb59823"},
  "ash_postgres": {:hex, :ash_postgres, "1.5.26", "23e031f3570bf436455224d58e9230b2290c7e7b4b03c69cef476d716acff0e3", [:mix], [{:ash, ">= 2.20.3 and < 3.0.0-0", [hex: :ash, repo: "hexpm", optional: false]}, {:ecto, "~> 3.9", [hex: :ecto, repo: "hexpm", optional: false]}, {:ecto_sql, "~> 3.9", [hex: :ecto_sql, repo: "hexpm", optional: false]}, {:jason, "~> 1.0", [hex: :jason, repo: "hexpm", optional: false]}, {:postgrex, ">= 0.0.0", [hex: :postgrex, repo: "hexpm", optional: false]}], "hexpm", "2b0e14f21818e32173e31f301e93e70295309bec71ea80de9817e6fa2df4ff38"},
  "ash_state_machine": {:hex, :ash_state_machine, "0.2.2", "baabbcbf6580c157d136c522f8eac76f2096ca02b1831e72c5898647e83029c6", [:mix], [{:ash, "~> 2.7", [hex: :ash, repo: "hexpm", optional: false]}, {:spark, ">= 1.1.22", [hex: :spark, repo: "hexpm", optional: false]}], "hexpm", "efd38070d217539cd36a2a5b494063a0cb94a1d80cfa5cde627dc0ee2cce5657"},
 ```

I'm going through each upside down

Reply:
> Actually... 🤔 this shouldn't be an issue if Ash is also upgraded

But 0.28.1 does not point on newer Ash that adds that, right? Otherwise, it would be a bug. It does not require that newer version so it was a breaking change. No action needed. Just leaving trace for others.

Reply:
Ah, right, this updates the `mix.lock` but not the `mix.exs`.

Reply:
Ah, yeah I see what you're saying. And some later version does eventually bump the requirement, so nothing to be done. But good for others to see if they have the same issue. Apologiies I didn't notice this when it happened :)